packages/core/src/stream-loading/state-manipulator.ts

```diff
@@ -1,27 +1,266 @@
-import { AppliableStreamLog, StreamState } from '@ceramicnetwork/common'
+import {
+  AppliableStreamLog,
+  CommitType,
+  Context,
+  DiagnosticsLogger,
+  LogEntry,
+  Stream,
+  StreamHandler,
+  StreamState,
+  StreamUtils,
+  UnappliableStreamLog,
+} from '@ceramicnetwork/common'
+import { HandlersMap } from '../handlers-map.js'
+import { LogSyncer } from './log-syncer.js'
+
+/**
+ * @param throwOnInvalidCommit - if true, throws if there is an error applying a commit, otherwise
+ *   returns the state that was built so far.
+ */
+interface ApplyFullLogOpts {
+  throwOnInvalidCommit: boolean
+}
+
+/**
+ * @param throwOnInvalidCommit - if true, throws if there is an error applying a commit, otherwise
+ *   returns the state that was built so far.
+ * @param throwIfStale - if true, throws if the log to apply does not build directly on top of
+ *   the existing state.
+ * @param throwOnConflict - if true, throws if the log to apply is rejected by conflict resolution.
+ */
+interface ApplyLogToStateOpts {
+  throwOnInvalidCommit: boolean
+  throwIfStale: boolean
+  throwOnConflict: boolean
```

Comment:
Do these default to `true` or `false` when applying the logic please?
If they default to `true` and must be explicitly set to `false`, could we inverse the logic please? Ex:
```ts
allowInvalidCommit: boolean
allowStale: boolean
allowConflict: boolean
```
With these flags having to be set explicitly to `true` not to throw? I think that would make it clearer the default behavior is to be strict, unless explicitly set?

Reply:
they don't actually default to anything, it is required for the caller to pass all 3 in explicitly.  There isn't really a clear case for which behavior should be the "default" - they are all set differently in different contexts.  E.g. `throwOnInvalidCommit` is true for user-initiated writes coming in via the http api, but false for writes that come in via pubsub.